tests/generation/FileGlobs.py

```diff
@@ -96,3 +96,36 @@ def test_get_project_files_gitignore(self):
 
         os.remove(".gitignore")
         os.chdir(orig_cwd)
+
+
+    def test_get_project_files_ci_mode(self):
+        orig_cwd = os.getcwd()
+        os.chdir(os.path.dirname(os.path.realpath(__file__)))
+        os.makedirs("file_globs_ci_testfiles", exist_ok=True)
+        os.chdir("file_globs_ci_testfiles")
+
+        os.makedirs("src", exist_ok=True)
+        os.makedirs("ignore_dir", exist_ok=True)
+        touch("file.c", "src")
+        touch("root.c")
+        touch("src.c", "ignore_dir")
+        touch("src.js", "ignore_dir")
+
+        with suppress_stdout():
+            res, _ = get_project_files(self.log_printer, self.printer, os.getcwd()
+                , True)
+            self.assertIn(os.path.join(os.getcwd(), "src", "file.c"), res)
+            self.assertIn(os.path.join(os.getcwd(), "root.c"), res)
+            self.assertIn(os.path.join(os.getcwd(), "ignore_dir" 
+                + os.sep + "src.c"), res)
+            self.assertIn(os.path.join(os.getcwd(), "ignore_dir" 
+                + os.sep + "src.js"), res)
+
+        os.chdir(orig_cwd)
+
+
+def touch(file_name, dir_name=None):
```

Comment:
this order doesn't make sense, rather do `*args* and then do an `os.path.join(*args)` and open on that, no if needed anymore :)